app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawerTest.kt

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @Test
+    fun `calculateLayoutParams for single *none* UTC lecture sets top margin 0 (its the first lecture in all rooms, so on the top)`() {
+        val startTime = 10 * 60 // 10:00am
+        val lectures = listOf(createLecture(_roomIndex = 0, time = startTime))
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lectureParams = layoutParams[lectures.first()]
+
+        assertMargins(lectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams for consecutive lecture sets margins based on gap duration`() {
+        val startTime1 = 10 * 60 // 10:00am
+        val duration1 = 45
+        val gapMinutes = 15
+        val startTime2 = startTime1 + duration1 + gapMinutes // 11:00am
+
+        val lecture1 = createLecture(_roomIndex = 0, date = conferenceDate, time = startTime1, _duration = duration1)
+        val lecture2 = createLecture(_roomIndex = 0, date = conferenceDate, time = startTime2)
+        val lectures = listOf(lecture1, lecture2)
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lecture1Params = layoutParams[lecture1]
+        val secondLectureParams = layoutParams[lecture2]
+
+        assertMargins(lecture1Params, 0, gapMinutes)
+        assertMargins(secondLectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams for consecutive lecture in another room sets top margin based on gap duration`() {
+        /*
+                         room 0             room 1
+                   +---------------------------------------+
+            10:00  +-------------------+                   |  +
+                   |                   |                   |  |
+                   |     lecture 0     |                   |  |
+                   |                   |                   |  | marginTop
+            10:45  +-------------------+                   |  |
+                   |                   |                   |  |
+            11:00  |                   +-------------------+  +
+                   |                   |                   |
+                   |                   |    lecture 1      |
+                   |                   |                   |
+
+        * lecture 1 follows directly lecture 0, but in another room, hence the margin includes height of lecture 0.
+        */
+        val duration1 = 45
+        val startTime1 = 10 * 60 // 10:00am
+        val startTime2 = startTime1 + duration1 + 15 // 11:00am
+
+        val lecture1 = createLecture(_roomIndex = 0, date = conferenceDate, time = startTime1, _duration = duration1)
+        val lecture2 = createLecture(_roomIndex = 1, date = conferenceDate, time = startTime2)
```

Comment:
:art: About understanding test easily: When I study all the tests then the initial calculation appears kind of an overhead for me to read. I understand that the calculations above is needed. However, my first thought is: When does a lecture start and when does it end? This feels intuitive. Maybe something like this might make it a bit easier to gasp:

``` kotlin
val lecture1 /* 10:00-10:45 */ = createLecture(_roomIndex = 0, date = conferenceDate, time = startTime1, _duration = duration1)
val lecture2 /* 11:00-..... */ = createLecture(_roomIndex = 1, date = conferenceDate, time = startTime2)
```
With this I would initially ignore the above calculation and everything after the assignment and just read the code below. Only when I want to understand how the start time was calculated I would read the above.

Reply:
I decided to comment at the startTime variable definitions, feels more natural / common to me.

Reply:
Okay. A bit over :brain: work when reading the tests - not optimal - but maybe its healthy ^^.